kafka_consumer/datadog_checks/kafka_consumer/kafka_consumer.py

```diff
@@ -70,15 +68,104 @@ def check(self, _):
             )
 
         # Report the metrics
-        self.client.report_highwater_offsets(self._context_limit)
-        self.client.report_consumer_offsets_and_lag(self._context_limit - len(self.client._highwater_offsets))
+        # Expected format: {(consumer_group, topic, partition): offset}
+        self._consumer_offsets = self.client.get_consumer_offsets_dict()
+        # Expected format: {(topic, partition): offset}
+        self._highwater_offsets = self.client.get_highwater_offsets_dict()
```

Comment:
I'm a little concerned about this change here. The consumer_offsets and highwater_offsets are set to `{}` on every check run. However, in the `kafka_python_client`, the consumer_offsets and highwater_offsets dicts are initialized only when the class is initialized. The dicts never get reset in the client. I wonder if that'll change the behavior of the check or cause the dicts to keep growing larger

Reply:
Yeah that's a good point, we probably shouldn't be keeping a `self._consumer_offsets` or `self._highwater_offsets` value in the `KafkaPythonClient` since the client stays alive for the entire duration. I think instead maybe we can initialize `*_offsets` in each respective `get_*_offsets_dict()`? Another possibility is we could have a function to "reset" the values of the offsets before each check run, although that would be less pretty.

Reply:
Thanks, I'll try both of those options out and see which works better

Reply:
I ended up adding a `reset_offsets` function to the client which the check will call at the beginning of every check run